src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -938,6 +943,17 @@ def get_environment_client(self):
         else:
             return ManagedEnvironmentPreviewClient
 
+    def set_up_runtime(self):
+        runtime_option = infer_runtime_option(self.get_argument_runtime(), self.get_argument_enable_java_metrics())
+        runtime_def = None
+        if runtime_option == RUNTIME_JAVA:
+            runtime_java_def = RuntimeJavaModel
+            runtime_java_def["enableMetrics"] = self.get_argument_enable_java_metrics() or False
+            runtime_def = {
+                "java": runtime_java_def
+            }
+        safe_set(self.containerapp_def, "properties", "configuration", "runtime", value=runtime_def)
```

Comment:
If `runtime_option` is not Java, the value of `runtime_def` will be `None`.
Is it excepted?

If it is expected, I think the `safe_set` logic should under the `if` at line 949.

Reply:
If we decide to extend `--runtime` in the future, such as adding `--runtime=dotnet`, we would simply introduce an additional `if` branch to generate the corresponding `runtime_def`.

If `safe_set None` is not acceptable, shall we add another `if` branch for `RUNTIME_GENERIC` to short circuit the logic?